execute/exectypes/observation.go

```diff
@@ -18,6 +18,68 @@ type MessageObservations map[cciptypes.ChainSelector]map[cciptypes.SeqNum]ccipty
 // must be encoding according to the destination chain requirements with typeconv.AddressBytesToString.
 type NonceObservations map[cciptypes.ChainSelector]map[string]uint64
 
+// TokenDataObservations contain token data for messages organized by source chain selector and sequence number.
+// There could be multiple tokens per a single message, so MessageTokensData is a slice of TokenData.
+// TokenDataObservations are populated during the Observation phase and depends on previously fetched
+// MessageObservations details and the `tokenDataProcessors` configured in the ExecuteOffchainConfig.
+// Content of the MessageTokensData is determined by the TokenDataProcessor implementations.
+//   - if Message doesn't have any tokens, TokenData slice will be empty.
```

Comment:
Suppose ChainSelector `chain` and SeqNum `seq` is a message that does not have token data. Is there no `MessageTokenData` object in the map? In that case `tokenObs[chain][seq].IsReady()` is false even though we're not waiting for data.

Reply:
`MessageTokenData` has to be always there, that's gonna be a requirement here. Therefore client doesn't have to handle nils or other weird edge cases. We move this responsibility to the processor to init everything properly, even messages that don't have tokens. In this case, it should be populated with an empty `MessageTokensData` which is always `IsReady=true` in that case 

```go
func (mtd MessageTokensData) IsReady() bool {
	for _, td := range mtd.TokenData {
		if !td.IsReady() {
			return false
		}
	}
	return true
}
```

Reply:
I think comment explains that, no? `TokenData slice will be empty` and for empty TokenData we return `IsReady=true`